### PR TITLE
NodeTypeResolver: remove unnecessary check

### PR DESCRIPTION
**Pull request by staabm**
running `vendor/bin/phpunit` on rector-src 

before this PR: 177.64s
after this PR: 172.23s

----

after https://github.com/rectorphp/rector-src/pull/3627 is merged, there should be enough room to merge this without a need to refactor because of too much complexity

### CODE REVIEW

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
CI needs to pass first, could you handle it?

File: packages/NodeTypeResolver/NodeTypeResolver.php [hunk 1]
```diff
@@ -365,7 +365,7 @@ private function isObjectTypeOfObjectType(ObjectType $resolvedObjectType, Object
             }
         }
 
-        return $resolvedClassReflection->isSubclassOf($requiredClassName);
```
Comment (staabm) [inline packages/NodeTypeResolver/NodeTypeResolver.php:368]: the previous `$resolvedObjectType->isInstanceOf($requiredClassName)` check already does the internally. so we don't need todo it here again

https://github.com/phpstan/phpstan-src/blob/f6c0f39c8ba03d20f55d7ae32cd4c62b811aa19d/src/Type/ObjectType.php#L1294-L1296

**Comment by staabm** [comment]
@TomasVotruba this one is good to go

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @staabm

**Pull request merged**